aws/resource_aws_licensemanager_license_configuration.go

```diff
@@ -0,0 +1,189 @@
+				Optional: true,
+			},
+			"license_count_hard_limit": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"license_counting_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					licensemanager.LicenseCountingTypeVCpu,
+					licensemanager.LicenseCountingTypeInstance,
+					licensemanager.LicenseCountingTypeCore,
+					licensemanager.LicenseCountingTypeSocket,
+				}, false),
+			},
+			"license_rules": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsLicenseManagerLicenseConfigurationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).licensemanagerconn
+
+	opts := &licensemanager.CreateLicenseConfigurationInput{
+		LicenseCountingType: aws.String(d.Get("license_counting_type").(string)),
+		Name:                aws.String(d.Get("name").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		opts.Description = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("license_count"); ok {
+		opts.LicenseCount = aws.Int64(int64(v.(int)))
+	}
+
+	if v, ok := d.GetOk("license_count_hard_limit"); ok {
+		opts.LicenseCountHardLimit = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("license_rules"); ok {
+		opts.LicenseRules = expandStringList(v.([]interface{}))
+	}
+
+	if v, ok := d.GetOk("tags"); ok && len(v.(map[string]interface{})) > 0 {
+		opts.Tags = tagsFromMapLicenseManager(v.(map[string]interface{}))
+	}
+
+	log.Printf("[DEBUG] License Manager license configuration: %s", opts)
+
+	resp, err := conn.CreateLicenseConfiguration(opts)
+	if err != nil {
+		return fmt.Errorf("Error creating License Manager license configuration: %s", err)
+	}
+	d.SetId(*resp.LicenseConfigurationArn)
+	return resourceAwsLicenseManagerLicenseConfigurationRead(d, meta)
+}
+
+func resourceAwsLicenseManagerLicenseConfigurationRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).licensemanagerconn
+
+	resp, err := conn.GetLicenseConfiguration(&licensemanager.GetLicenseConfigurationInput{
+		LicenseConfigurationArn: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if isAWSErr(err, licensemanager.ErrCodeInvalidParameterValueException, "") {
+			log.Printf("[WARN] License Manager license configuration (%s) not found, removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading License Manager license configuration: %s", err)
+	}
+
+	d.Set("description", resp.Description)
+	d.Set("license_count", resp.LicenseCount)
+	d.Set("license_count_hard_limit", resp.LicenseCountHardLimit)
+	d.Set("license_counting_type", resp.LicenseCountingType)
+	d.Set("license_rules", flattenStringList(resp.LicenseRules))
```

Comment:
Nit: This looks okay, but in general if its a `TypeList`, `TypeSet`, or `TypeMap` we should be adding the error checking. 👍